Docs: Update ecctl usage examples (#277)
elastic/ecctl: docs/ecctl-getting-started.asciidoc

[id="{p}-examples"]
== Usage examples

> Once you've <<{p}-installing,installed>> and <<{p}-configuring,configured>> {p} you're ready to take it for a test drive. Here are some examples for you to try, demonstrating basic Create, Read, Update and Delete operations using {p}:

Reviewer: ```suggestion
Once you've <<{p}-installing,installed>> and <<{p}-configuring,configured>> {p} you're ready to take it for a test drive. Try these examples with basic Create, Read, Update and Delete operations using {p}:
```

* <<{p}-example-list-deployments,List deploymments>>
* <<{p}-example-create-deployment,Create a deployment>>
* <<{p}-example-update-deployment,Update a deployment>>
* <<{p}-example-delete-deployment,Delete a deployment>>

> If you would like to compare the {p} commands against their API equivalents, see the {cloud}/ec-api-examples.html[API examples].

Reviewer: ```suggestion
To compare the {p} commands against their API equivalents, see the {cloud}/ec-api-examples.html[API examples].
```

[id="{p}-example-list-deployments"]
=== List deployments

> As a first example of using {p}, let's run the <<ecctl_deployment_list,{p} deployment list>> command to retrieve information about existing deployments. This is a good way to confirm that {p} is configured correctly. If you have any deployments already created, they'll be listed in the results.

Reviewer: ```suggestion
As a first example of using {p}, run the <<ecctl_deployment_list,{p} deployment list>> command to retrieve information about existing deployments. This is a good way to check if {p} is configured correctly and if you have any deployments already created.
```

[source, sh]
--
ecctl deployment list
--

[source, sh]
--
ID                                 NAME            ELASTICSEARCH                      KIBANA                             APM                                APPSEARCH
00be03849b6a49c1a6541e3ccb5958d2   marvin          00be03849b6a49c1a6541e3ccb5958d2   266e456acf257588a9cde6fb4569d4a0   78c096c22e12408b878083b2d5ff6bcf   -
147cdeace6404c3e4b5018e1401647e4   biggerdata      147cdeace6404c3e4b5018e1401647e4   443a9df7b33952f45921c5823cbad4bc   4678ce52d45547e463455ede663cb4a4   -
--

[id="{p}-example-create-deployment"]
=== Create a deployment

> Let's create a basic deployment. Elasticsearch Service supports {cloud}/ec-getting-started-templates.html[deployment templates], which pre-configure the Elastic Stack components in your deployment to best suit your particular use case. For this example, we'll use Google Cloud Platform (GCP) to host the deployment in region `US Central 1 (Iowa)`. For details about what deployment options are available by platform, see our list of {cloud}/ec-regions-templates-instances.html[available regions, deployment templates and instance configurations].

Reviewer: ```suggestion
Let's create a basic deployment. Elasticsearch Service supports {cloud}/ec-getting-started-templates.html[deployment templates], which pre-configure the Elastic Stack components in your deployment to best suit your particular use case. For this example, use Google Cloud Platform (GCP) to host the deployment in region `US Central 1 (Iowa)`. To know which deployment options are available by platform, see {cloud}/ec-regions-templates-instances.html[available regions, deployment templates and instance configurations].
```

[source, sh]
--
ecctl deployment create -f create-deployment.json
--

[source, sh]
--
{
  "created": true,
  "id": "7229888e7bf8350c7e4d07d7374171c0",
  "name": "My first ecctl deployment",
  "resources": [
    {
      "cloud_id": "My_first_ecctl_deployment:dXMtY2VudHJhbDEuZ2NwLmZvdW5kaXQubm8kYjFlZWVjOGQ0YWVlNGY3ZDgxNTM2Zjc1ZjZhN2Y1MDgkM2ViZTAzNmI0NDhkNDc3Y2E2ZTJjZTQ5NmE4ZDQ5ODA=",
      "credentials": {
        "password": "qIJuMbL8BFfgoNq5bpjQhDRC",
        "username": "elastic"
      },
      "id": "b1eeec8d4aee4f7d81536f75f6a7f508",
      "kind": "elasticsearch",
      "ref_id": "main-elasticsearch",
      "region": "gcp-us-central1"
    },
    {
      "elasticsearch_cluster_ref_id": "main-elasticsearch",
      "id": "3ebe036b448d477ca6e2ce496a8d4980",
      "kind": "kibana",
      "ref_id": "main-kibana",
      "region": "gcp-us-central1"
    },
    {
      "elasticsearch_cluster_ref_id": "main-elasticsearch",
      "id": "5a03472f6dfe4f17acbe62622823b9cb",
      "kind": "apm",
      "ref_id": "main-apm",
      "region": "gcp-us-central1",
      "secret_token": "zfufcfe15eCVJk78b5"
    }
  ]
}
--

> The response indicates that the request was submitted successfully. It includes the `elastic` user password, which you can use to log in to Kibana or to access the Elasticsearch REST API. Make a note of the deployment ID, which we will use in the next example.

Reviewer: ```suggestion
The response indicates that the request was submitted successfully. It includes the `elastic` user password, which you can use to log in to Kibana or to access the Elasticsearch REST API. Make a note of the deployment ID, which you will use in the next example.
```

[id="{p}-example-update-deployment"]
=== Update a deployment

> Now that you have used {p} to create a deployment, let's update it. We will scale it up, increasing the size of the Elasticsearch data nodes from 4096 to 8192 MB.

Reviewer: ```suggestion
Now that you have used {p} to create a deployment, you can scale it up, by increasing the size of the Elasticsearch data nodes from 4096 to 8192 MB.
```

  "resources": [
    {
      "cloud_id": "My_second_ecctl_deployment:dXMtY2VudHJhbDEuZ2NwLmZvdW5kaXQubm8kYjc0OWU2ZWExN2Y4NDg5Yzg4Y2UyOTVjZTA4ZDVjNWUkNTliZWJiYjE3ZmFkNDk2MWEwMmNkMDRmNzYyOWYxMTk=",
      "id": "b749e6ea17f8489c88ce295ce08d5c5e",
      "kind": "elasticsearch",
      "ref_id": "main-elasticsearch",
      "region": "gcp-us-central1"
    },
    {
      "elasticsearch_cluster_ref_id": "main-elasticsearch",
      "id": "59bebbb17fad4961a02cd04f7629f119",
      "kind": "kibana",
      "ref_id": "main-kibana",
      "region": "gcp-us-central1"
    },
    {
      "elasticsearch_cluster_ref_id": "main-elasticsearch",
      "id": "1ec19461253c4175a2cea6b3ccc399a8",
      "kind": "apm",
      "ref_id": "main-apm",
      "region": "gcp-us-central1"
    }
  ]
}
--

[id="{p}-example-delete-deployment"]
=== Delete a deployment

In this last example, you can use the <<ecctl_deployment_shutdown,{p} deployment shutdown>> command to delete the deployment that you created.

[source, json]
--
ecctl deployment shutdown $DEPLOYMENT_ID
--
* `$DEPLOYMENT_ID` is the ID for the deployment that was created in the previous <<{p}-example-create-deployment,create a deployment>> example.

> On running this and other destructive commands, {p} prompts you with a confirmation message. You can use the `--force` option to skip the confirmation step, which can be useful if you are using {p} for automation.

Reviewer: ```suggestion
On running this and other destructive commands, {p} prompts you with a confirmation message. Use the `--force` option to skip the confirmation step, if you are using {p} for automation.
```

> There’s much more you can use {p} for. You can explore its various commands by issuing `ecctl <command> <help>` to see the different options that we support.

Reviewer: ```suggestion
To see the different options that {p} supports, run `ecctl <command> <help>`.
```
